NDTensors/src/default_storage.jl

```diff
@@ -0,0 +1,23 @@
+## This is a fil which specifies the default storage type provided some set of parameters
+## The parameters are the element type and storage type
+default_datatype(eltype::Type) = Vector{eltype}
+default_datatype() = Vector{default_eltype()}
```

Comment:
I would write this as:
```julia
default_datatype() = default_datatype(default_eltype())
```
Even though these are short functions, I find it is best to think about "implementation functions" that are at the bottom that do the work, and processing functions that process the inputs with the goal of calling the "implementation functions". Here it is useful since there will only be one point in the code where `Vector` is actually defined as the default data type (and therefore would be easier to customize, if we decide on that direction).